pylint/checkers/similar.py

```diff
@@ -43,15 +69,294 @@
 from pylint.reporters.ureports.nodes import Table
 from pylint.utils import decoding_stream
 
+DEFAULT_MIN_SIMILARITY_LINE = 4
+
 REGEX_FOR_LINES_WITH_CONTENT = re.compile(r".*\w+")
 
+# Index defines a location in a LineSet stripped lines collection
+Index = NewType("Index", int)
+
+# LineNumber defines a location in a LinesSet real lines collection (the whole file lines)
+LineNumber = NewType("LineNumber", int)
+
+
+# LineSpecifs holds characteristics of a line in a file
+class LineSpecifs(NamedTuple):
+    line_number: LineNumber
+    text: str
+
+
+# Links LinesChunk object to the starting indices (in lineset's stripped lines)
+# of the different chunk of lines that are used to compute the hash
+HashToIndex_T = Dict["LinesChunk", List[Index]]
+
+# Links index in the lineset's stripped lines to the real lines in the file
+IndexToLines_T = Dict[Index, "SuccessiveLinesLimits"]
+
+
+class CplSuccessiveLinesLimits:
+    """
+    This class holds a couple of SuccessiveLinesLimits objects, one for each file compared,
+    and a counter on the number of common lines between both stripped lines collections extracted
+    from both files
+    """
+
+    __slots__ = ("first_file", "second_file", "effective_cmn_lines_nb")
+
+    def __init__(
+        self,
+        first_file: "SuccessiveLinesLimits",
+        second_file: "SuccessiveLinesLimits",
+        effective_cmn_lines_nb: int,
+    ) -> None:
+        self.first_file = first_file
+        self.second_file = second_file
+        self.effective_cmn_lines_nb = effective_cmn_lines_nb
+
+
+# Links the indices ot the starting line in both lineset's stripped lines to
+# the start and end lines in both files
+CplIndexToCplLines_T = Dict["LineSetStartCouple", CplSuccessiveLinesLimits]
+
+
+class LinesChunk:
+    """
+    The LinesChunk object computes and stores the hash of some consecutive stripped lines of a lineset.
+    """
+
+    __slots__ = (
+        "_fileid",
+        "_index",
+        "_hash",
+    )
```

Comment:
```suggestion
    __slots__ = ("_fileid", "_index", "_hash")
```
If you remove the trailing comma, this will be formatted a bit better by black.

Reply:
Done